deepspeed/runtime/utils.py

```diff
@@ -32,6 +33,11 @@ def check_using_norm(self, norm_group):
                                          op=torch.distributed.ReduceOp.MAX,
                                          group=self.mpu.get_model_parallel_group())
             overflow = overflow_gpu[0].item()
+        else:
+            cuda_overflow = torch.cuda.FloatTensor([overflow])
+            dist.all_reduce(cuda_overflow, op=torch.distributed.ReduceOp.MAX)
+            dist.barrier()
+            overflow = cuda_overflow[0].item()
```

Comment:
I don't think we need to do this. This function is used in fp16/fused_optimizer.py and fp16/unfused_optimizer.py as well as in 1bit-adam. If we really need to do this for 1bit-adam let's have an `elif` here and pass a parameter into `check_using_norm` to do the reduction. I believe in standard fp16 training with deepspeed at this point all ranks will have already all-reduced their gradients which means everyone will have an overflow or no one will have an overflow.

Reply:
Need to check with Hank on this one. I did not add these lines. I will update this when I hear back.

Reply:
Sounds good, let's merge this into staging v2 as soon as we get this comment settled.